volatility3/framework/plugins/windows/netscan.py

```diff
@@ -142,76 +142,94 @@ def determine_tcpip_version(cls, context: interfaces.context.ContextInterface, l
-                (6, 3, 9600): "netscan-win81-x64",
-                (10, 0, 10240): "netscan-win10-x64",
-                (10, 0, 10586): "netscan-win10-x64",
-                (10, 0, 14393): "netscan-win10-x64",
-                (10, 0, 15063): "netscan-win10-15063-x64",
-                (10, 0, 16299): "netscan-win10-16299-x64",
-                (10, 0, 17134): "netscan-win10-17134-x64",
-                (10, 0, 17763): "netscan-win10-17763-x64",
-                (10, 0, 18362): "netscan-win10-18362-x64",
-                (10, 0, 18363): "netscan-win10-18363-x64",
-                (10, 0, 19041): "netscan-win10-19041-x64"
+                (6, 0, 6000, 0): "netscan-vista-x64",
+                (6, 0, 6001, 0): "netscan-vista-sp12-x64",
+                (6, 0, 6002, 0): "netscan-vista-sp12-x64",
+                (6, 0, 6003, 0): "netscan-vista-sp12-x64",
+                (6, 1, 7600, 0): "netscan-win7-x64",
+                (6, 1, 7601, 0): "netscan-win7-x64",
+                (6, 1, 8400, 0): "netscan-win7-x64",
+                (6, 2, 9200, 0): "netscan-win8-x64",
+                (6, 3, 9600, 0): "netscan-win81-x64",
+                (6, 3, 9600, 19935): "netscan-win81-19935-x64",
+                (10, 0, 10240, 0): "netscan-win10-x64",
+                (10, 0, 10586, 0): "netscan-win10-x64",
+                (10, 0, 14393, 0): "netscan-win10-x64",
+                (10, 0, 15063, 0): "netscan-win10-15063-x64",
+                (10, 0, 16299, 0): "netscan-win10-16299-x64",
+                (10, 0, 17134, 0): "netscan-win10-17134-x64",
+                (10, 0, 17763, 0): "netscan-win10-17763-x64",
+                (10, 0, 18362, 0): "netscan-win10-18362-x64",
+                (10, 0, 18363, 0): "netscan-win10-18363-x64",
+                (10, 0, 19041, 0): "netscan-win10-19041-x64"
             }
 
-        # special use case: Win10_18363 is not recognized by windows.info as 18363
+        # we do not need to check for tcpip's specific FileVersion in every case
+        tcpip_mod_version = 0 # keep it 0 as a default
+
+        # special use cases
+
+        # Win10_18363 is not recognized by windows.info as 18363
         # because all kernel file headers and debug structures report 18363 as
         # "10.0.18362.1198" with the last part being incremented. However, we can use
         # os_distinguisher to differentiate between 18362 and 18363
         if vers_minor_version == 18362 and is_18363_or_later:
             vollog.debug("Detected 18363 data structures: working with 18363 symbol table.")
             vers_minor_version = 18363
 
+        # Win 8 SP 1 also may have different structures based on specific tcpip.sys version
+        if (nt_major_version, nt_minor_version, vers_minor_version) == (6, 3, 9600):
+            vollog.debug("Requiring further version inspection due to OS version by checking tcpip.sys's FileVersion header")
+            physical_layer_name = context.layers[layer_name].config.get('memory_layer', None)
```

Comment:
This is intel specific.  We haven't yet got a good way of stepping just below the virtual layer, but adding in a comment mentioning that it will need adapting if other arches come along might be handy.

Reply:
I've added an info comment.